cvat-ui/src/components/profile-page/profile-content.tsx

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { CombinedState } from 'reducers';
+
+import Form from 'antd/lib/form';
+import Card from 'antd/lib/card';
+import { Col, Row } from 'antd/lib/grid';
+import Input from 'antd/lib/input';
+import Button from 'antd/lib/button';
+
+import { updateUserAsync } from 'actions/auth-actions';
+
+function ProfileContent(): JSX.Element {
+    const user = useSelector((state: CombinedState) => state.auth.user);
+    const dispatch = useDispatch();
+    const [form] = Form.useForm();
+    const [hasChanges, setHasChanges] = useState(false);
+
+    const initialValues = {
+        username: user?.username,
+        email: user?.email,
+        firstName: user?.firstName,
+        lastName: user?.lastName,
+    };
+
+    useEffect(() => {
+        form.setFieldsValue(initialValues);
+    }, [user, form]);
+
+    const onValuesChange = (changedValues: any, allValues: any): void => {
+        const currentEditableValues = {
+            firstName: allValues.firstName,
+            lastName: allValues.lastName,
+        };
+        const initialEditableValues = {
+            firstName: initialValues.firstName,
+            lastName: initialValues.lastName,
+        };
+
+        const hasFormChanges = JSON.stringify(currentEditableValues) !== JSON.stringify(initialEditableValues);
+        setHasChanges(hasFormChanges);
+    };
+
+    const onFinish = async (values: any): Promise<void> => {
+        if (user && hasChanges) {
```

Comment:
Do we really need `onValuesChange` only to call `setHasChanges`? Why not calculate it here?